### PR TITLE
E1687. Instructor account creation over the web.

### PR DESCRIPTION
**Pull request by srikarpotta**

### CODE REVIEW

File: app/controllers/instructions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class InstructionsController < ApplicationController
```
Comment (yangsong8) [inline app/controllers/instructions_controller.rb]: Why this controller is necessary?

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -71,6 +78,15 @@ def list
     @letters = ('A'..'Z').to_a
   end
 
+  def review
```
Comment (yangsong8) [inline app/controllers/users_controller.rb]: I dont think it's a good idea to have a review method in users_controller.

**Comment by yangsong8** [comment]
The comment below is for grading purpose only. Do not change your code unless your are required to

File: app/views/users/request_new.html.erb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<h1>Request new user</h1>
+
+<%= form_tag :action => 'request_user_create' do %>
+  <p><label for="role_id">Role</label><br/>
+    <%= collection_select 'user', 'role_id', @all_roles, :id, :name, {:selected => @rolename.id} %></p>
+
+  <%= render :partial => 'user', :locals => { :new_user => true, :request_user => true } %>
+  <%= recaptcha_tags %>
```
Comment (yangsong8) [inline app/views/users/request_new.html.erb]: I got an error caused by this line when I deploy your code on my local machine.

File: app/views/mailer/request_user_message.html.erb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+</head>
+<body>
+Hi <%= @first_name %>,<br />
+<br />
+Below user requested to sign up. Review by going to 'Manage/Users/Review Requested Users'.<br />
```
Comment (yangsong8) [inline app/views/mailer/request_user_message.html.erb]: Cannot find this from UI.

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (chaitanya91k) [inline app/controllers/users_controller.rb]: Any suggestions on where to move this method. Should it be under requested_users_controller.rb?

File: app/controllers/instructions_controller.rb (hunk 1 above)
Comment (chaitanya91k) [inline app/controllers/instructions_controller.rb]: hmm yes. Thanks for pointing out. As there are no additional functionality in this, we can remove this.

**Comment by chaitanya91k** [comment]
For this -- **In the video, super admins can user menu item manage->user->Review Requested Users. But I do not see this in my local system.**

The cache in roles for super_Administrator in roles table needs to be refreshed. I could not figure out a clean way to refresh the cache. 
Well I deleted the cache from database roles table for super_Administrator and then reloaded the cache by logging in as a super_administrator (I had to comment few lines of code in models/menu.rb to load the cache)(Although I am sure there is a neat way to refresh the cache, I am not aware of).

**In the user/request_new page, there is no place for reason field in "Requested_user". In this the reason that the user wants to user Expertiza, or the reason that we reject the request? (I would miss something, but I didn't see this field is used either way.)**

Reason field was intended to be used bu Super_Administrator to log why they have rejected users request.
This is completely changeable based on what is requirement.   

**In the user/request_new page, a user has to select an institution from drop down.**

Yes, We are using the same view of new user. Shouldn't it be a drop down?
 We can change this if you need. If so, can you give more details on what exactly is required.

**No check on if the requested user name is available or no. We cannot allow an outside user to request an account "efg".**

We were planning to implement and thus added a TODO, But completely forgot about this. We can add this now if its ok.

**Pull request merged**